### PR TITLE
Split runtime into multiple files

### PR DESCRIPTION
**Pull request by j-f1**
This PR:

- splits the runtime TypeScript code up into several separate modules
- adds Rollup to compile that back into a single JS file
- uses Rollup to produce two bundles instead of one:
  - one that uses UMD to allow use with Node-style `require`, or by embedding it with a `<script>` tag and using `window.JavaScriptKit`
  - one that uses ES Modules
- refactor the codebase to use a `Memory` class that provides easy access to memory and the heap
- check for FinalizationRegistry availability in `SwiftClosureDeallocator` constructor rather than outside of it

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
Some tests are failing 👀 

```console

Error in Literal Conversion

MessageError(message: "Expect to be equal \"212205744152385\" and \"4746794007244308480\"", file: "/home/runner/work/JavaScriptKit/JavaScriptKit/IntegrationTests/TestSuites/Sources/PrimaryTests/main.swift", line: 26, column: 28)

Swift/ErrorType.swift:200: Fatal error: Error raised at top level: PrimaryTests.MessageError(message: "Expect to be equal \"212205744152385\" and \"4746794007244308480\"", file: "/home/runner/work/JavaScriptKit/JavaScriptKit/IntegrationTests/TestSuites/Sources/PrimaryTests/main.swift", line: 26, column: 28)

RuntimeError: unreachable
    at $ss17_assertionFailure__4file4line5flagss5NeverOs12StaticStringV_SSAHSus6UInt32VtF (wasm://wasm/02991d2a:wasm-function[2633]:0xccfd3)
    at swift_errorInMain (wasm://wasm/02991d2a:wasm-function[4819]:0x14b4ac)
    at main (wasm://wasm/02991d2a:wasm-function[1532]:0x76025)
    at main (wasm://wasm/02991d2a:wasm-function[29674]:0x622f25)
    at __main_void (wasm://wasm/02991d2a:wasm-function[29675]:0x622fa1)
    at __original_main (wasm://wasm/02991d2a:wasm-function[29672]:0x622ee6)
    at _start (wasm://wasm/02991d2a:wasm-function[45]:0x10884)
    at d.start (/home/runner/work/JavaScriptKit/JavaScriptKit/IntegrationTests/node_modules/@wasmer/wasi/lib/index.cjs.js:77:280)
    at startWasiTask (/home/runner/work/JavaScriptKit/JavaScriptKit/IntegrationTests/lib.js:47:10)
```

**Comment by j-f1** [comment]
It turned out to be an endianness issue, oof. Hoping this change fixes it 🤞

File: Runtime/src/closure-heap.ts
Comment (MaxDesiatov) [inline Runtime/src/closure-heap.ts:10]: nit: can we make this a multi-line string? No need to display it as multi-line to the user, just a matter of formatting in the source code.

File: Runtime/src/types.ts
Comment (MaxDesiatov) [inline Runtime/src/types.ts:102]: nit: needs a clarifying comment explaining why these are commented out

**Review by MaxDesiatov — APPROVED** [review]
Overall seems legit 👍 but I'd wait for at least one more approval from the team

**Review by kateinoigakukun — APPROVED** [review]

**Pull request merged**
